### PR TITLE
PPS Timing Detector: Updated mapping for 2018

### PR DESCRIPTION
**Pull request by nminafra**
We are updating the mapping of the PPS Timing Diamond detector to reflect the different cabling done during YETS.
During 2017 we had some bugs in the front-end electronics that were fixed with the Digitizer Board V3. Moreover, the new cabling is optimized to balance the occupancy in the HPTDCs to not saturate the read-out fifos.
The cabling will not change anymore during 2018.

### CODE REVIEW

**Comment by perrotta** [comment]
@nminafra @fabferro : cannot this be made available through a GT, as it was done for the other #22264 CTPPS pull request?

**Comment by fabferro** [comment]
@perrotta: you are right Andrea. Unfortunately what you ask had to be postponed due to higher priorities involving the same people (like @nminafra).

**Comment by slava77** [comment]
@cmsbuild please test

File: EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py [hunk 1]
```diff
@@ -68,10 +68,16 @@
     ),
     # 2017
     cms.PSet(
-      validityRange = cms.EventRange("292521:min - 999999999:max"),
+      validityRange = cms.EventRange("292521:min - 310000:max"),
       mappingFileNames = cms.vstring("CondFormats/CTPPSReadoutObjects/xml/mapping_timing_diamond_2017.xml"),
       maskFileNames = cms.vstring()
     )
+    # 2018
```
Comment (slava77) [inline EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py]: there is a comma missing above

**Comment by slava77** [comment]
ehm , the compilation log is more descriptive than the snippet provided by the bot:
```
Compiling src/EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py ...
  File "src/EventFilter/CTPPSRawToDigi/python/ctppsRawToDigi_cff.py", line 76
    cms.PSet(
      ^
SyntaxError: invalid syntax

```

@smuzaffar please check if the cms-bot error text selection can be improved in this case.
Thank you.

**Comment by nminafra** [comment]
@perrotta indeed, we had to test the new cabling on the detector first and we had quite a work to do with all the new detectors.
@slava77 sorry for the failed test! as said above, we had to rush a bit the things today.

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
@nminafra @fabferro : please tell us the schedule foreseen for moving the PPS Timing mappings into the DB. Given the experience you gained with the implementation of a similar move in the other PR I expect that it could be done before pre3: but let us know if this is not the case, and when you plan to do it if so

**Comment by fabferro** [comment]
@perrotta: we are installing the detectors these days. That means that the same people who should develop the code are now busy on hardware. I don't expect anything new about this subject for pre3 (next Monday) but I let @nminafra answer on this.

**Comment by nminafra** [comment]
@perrotta We'll try to do our best, but I cannot guarantee to be able to implement and test it by Monday.

**Comment by slava77** [comment]
> Given the experience you gained with the implementation of a similar move

@perrotta 
there is little similarity, actually.

If we migrate the mapping, it has to be done both for strips and diamonds.

**Comment by nminafra** [comment]
@perrotta we started the "procedure" but we'll not be able to do all the proper tests by today.
However, the updated mapping is very important from day 1 for the commissioning of the detectors during the first low(er) luminosity runs.

Thanks

**Comment by perrotta** [comment]
+1
- Added the mapping of the PPS Timing Diamond detector for the 2018 setup, which is accessed within the corresponding validity range

**Comment by arunhep** [comment]
@nminafra why do we need an xml file for the mapping? Don't we have a corresponding record in release?

**Comment by nminafra** [comment]
@arunhep We changed the mapping during YETS, therefore the old mapping is still needed for 2017 data. A similar procedure was used for Timing in 2017 and for Tracking in 2016 and 2017.
We plan a migration to db as soon as we have resources.

**Comment by arunhep** [comment]
@nminafra pardon my ignorance, is this CTPPSPixelDAQMappingRcd is different than the mapping discussed in this PR?

**Comment by nminafra** [comment]
@arunhep No problem at all. This PR involves only the timing detectors which are "similar" to Strip detectors. On the contrary, CTPPSPixel have a separate code, from unpacker to reco

**Comment by arunhep** [comment]
+1

**Comment by fabiocos** [comment]
@fabferro @nminafra I assume this is needed in the commissioning phase of the detector, could you please comment? When is the move to GT expected to happen?

**Comment by fabiocos** [comment]
@ggovi could you please review this PR?

**Comment by nminafra** [comment]
@fabiocos This is needed as soon as possible. Without the updated mapping we cannot use the DQM and it will slow down every commissioning operation.
Moving to the GT is foreseen after the commissioning: we have very limited resources and the same people were doing the installation and they'll be involved in the commissioning.

**Comment by ggovi** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**